packages/libs/error-stack/src/fmt/charset.rs

```diff
@@ -0,0 +1,128 @@
+use core::sync::atomic::{AtomicU8, Ordering};
+
+use crate::Report;
+
+/// The available supported charsets
+///
+/// Can be accessed through [`crate::fmt::HookContext::charset`], and set via
+/// [`Report::set_charset`].
+#[derive(Debug, Copy, Clone)]
+pub enum Charset {
+    /// Terminal of the user supports utf-8
+    ///
+    /// This is the default if no charset has been explicitly set.
+    Utf8,
+
+    /// Terminal of the user supports ASCII
+    Ascii,
+}
+
+impl Charset {
+    pub(super) fn load() -> Self {
+        // we assume that most fonts and terminals nowadays support Utf8, which is why this is
+        // the default
+        CHARSET_OVERRIDE
+            .load()
+            .map_or(Self::Utf8, |charset| charset)
+    }
+}
+
+/// Value layout:
+/// `0x00`: `Charset::Ascii`
+/// `0x01`: `Charset::Utf8`
+///
+/// all others: unset/none
+struct AtomicOverride(AtomicU8);
```

Comment:
`AtomicOverride` is very similar to `ColorPreference`, Is it possible to reuse the implementation?

Reply:
fixed in [de12b05](https://github.com/hashintel/hash/pull/1800/commits/de12b05ff5dae1663f87c3009307f929b38355bf)